### PR TITLE
Provisioner should ignore unconfirmed disputes

### PR DESCRIPTION
**Pull request by tdimitrov**
Fixes https://github.com/paritytech/polkadot/issues/6289

### CODE REVIEW

**Review by BradleyOlson64 — APPROVED** [review]
Had to read up on the provisioner to understand this one. Looks good!

File: node/core/provisioner/src/disputes/prioritized_selection/mod.rs [hunk 1]
```diff
@@ -139,6 +139,13 @@ where
 		onchain.len(),
 	);
 
+	// Filter out unconfirmed disputes. However if the dispute is already onchain - don't skip it.
+	// In this case we'd better push as much fresh votes as possible to bring it to conclusion faster.
+	let recent_disputes = recent_disputes
+		.into_iter()
+		.filter(|d| d.2.is_confirmed_concluded() || onchain.contains_key(&(d.0, d.1)))
```
Comment (eskimor) [inline node/core/provisioner/src/disputes/prioritized_selection/mod.rs:146]: Nice! Good reasoning. If it is already on chain, we also know it is good.

**Review by eskimor — APPROVED** [review]
:smiley:

**Comment by tdimitrov** [comment]
bot merge

**Pull request merged**